core/jvm/src/test/scala/scalaz/zio/RTSSpec.scala

```diff
@@ -963,6 +968,33 @@ class RTSSpec(implicit ee: ExecutionEnv) extends TestRuntime {
     } yield result must_=== 10
   }
 
+  def testInterruptStatusIsHeritable = unsafeRun {
+    for {
+      latch <- Promise.make[Nothing, Unit]
+      ref   <- Ref.make(InterruptStatus.interruptible)
+      _     <- ZIO.uninterruptible((ZIO.checkInterruptible(ref.set) *> latch.succeed(())).fork *> latch.await)
+      v     <- ref.get
+    } yield v must_=== InterruptStatus.uninterruptible
+  }
+
+  def testExecutorIsHeritable = unsafeRun {
+    for {
+      ref  <- Ref.make(Option.empty[internal.Executor])
+      exec = internal.Executor.fromExecutionContext(100)(scala.concurrent.ExecutionContext.Implicits.global)
+      _    <- IO.descriptor.map(_.executor).flatMap(e => ref.set(Some(e))).fork.lock(exec)
+      v    <- ref.get
+    } yield v must_=== Some(exec)
+  }
+
+  def testSupervisionIsHeritable = unsafeRun {
```

Comment:
I had this test that verifies it works "in practice", you can include it if you want (it passes):
```scala
def testSupervisingInheritance = {
    def forkAwaitStart[A](io: UIO[A]) =
      for {
        latch <- Promise.make[Nothing, Unit]
        _     <- (latch.succeed(()) *> io).fork
        _     <- latch.await
      } yield ()

    unsafeRun(
      (for {
        _     <- forkAwaitStart(forkAwaitStart(forkAwaitStart(IO.succeed(()))))
        fibs <- ZIO.children
      } yield fibs must have size 3).supervised
    )
  }
```

Reply:
Added it, thank you! 🙏